### PR TITLE
build both shared and static library

### PR DESCRIPTION
**Pull request by jeroen**
This enabled building a static library, in addition to a shared one. I need this myself for building standalone applications, but I think it is generally useful.

Thank you for your review.

### CODE REVIEW

File: src/lib/openjp2/CMakeLists.txt [hunk 1]
```diff
@@ -87,7 +87,9 @@ if(WIN32)
     add_definitions(-DOPJ_STATIC)
   endif()
 endif()
-add_library(${OPENJPEG_LIBRARY_NAME} ${OPENJPEG_SRCS})
+add_library(${OPENJPEG_LIBRARY_NAME} SHARED ${OPENJPEG_SRCS})
+add_library(openjp2_static STATIC ${OPENJPEG_SRCS})
```
Comment (rouault) [inline src/lib/openjp2/CMakeLists.txt]: Any reason not use ${OPENJPEG_LIBRARY_NAME} as well ? Generally both static and dynamic library have the same basename.

Comment (jeroen) [inline src/lib/openjp2/CMakeLists.txt]: cmake doesn't allow for duplicate target name I think

Comment (rouault) [inline src/lib/openjp2/CMakeLists.txt]: https://stackoverflow.com/questions/2152077/is-it-possible-to-get-cmake-to-build-both-a-static-and-shared-version-of-the-sam and https://datainfer.wordpress.com/2013/10/24/make-both-static-and-shared-libraries-in-one-build-with-cmake/ give one hint how to do that with set_target_properties(MyLibStatic PROPERTIES OUTPUT_NAME MyLib), but this one must be restricted to non-Windows platform, as Windows apparently need different names.

Comment (jeroen) [inline src/lib/openjp2/CMakeLists.txt]: Yes the PR already uses the `set_target_properties`.

**Comment by rouault** [comment]
Fails to build on Windows : http://my.cdash.org/buildSummary.php?buildid=1229211

**Comment by jeroen** [comment]
OK let me condition this on Non windows.

**Comment by jeroen** [comment]
Thanks, can you try again?

**Comment by jeroen** [comment]
Ping? This should be good to go. Thanks for reviewing!

**Pull request merged**